### PR TITLE
Add EKS Secret envelope encryption support

### PR DESCRIPTION
**Pull request by babilen5**
# PR o'clock

## Description

This adds support for configuring EKS clusters that utilise envelope
encryption for Secrets:

- https://aws.amazon.com/about-aws/whats-new/2020/03/amazon-eks-adds-envelope-encryption-for-secrets-with-aws-kms/

- https://aws.amazon.com/blogs/containers/using-eks-encryption-provider-support-for-defense-in-depth/

This addresses #767

### Checklist

- [X] Change added to CHANGELOG.md. All changes must be added and breaking changes and highlighted
- [X] CI tests are passing
- [X] README.md has been updated after any changes to variables and outputs. See https://github.com/terraform-aws-modules/terraform-aws-eks/#doc-generation

### CODE REVIEW

**Comment by barryib** [comment]
Thanks @babilen5 this is awesome. Can you please update your branch and resolve conflicts ?

**Comment by babilen5** [comment]
> Can you please update your branch and resolve conflicts ?

Done, @barryib.

File: variables.tf [hunk 1]
```diff
@@ -311,3 +311,12 @@ variable "eks_oidc_root_ca_thumbprint" {
   description = "Thumbprint of Root CA for EKS OIDC, Valid until 2037"
   default     = "9e99a48a9960b14926bb7f3b02e22da2b0ab7280"
 }
+
+variable "encryption_config" {
```
Comment (barryib) [inline variables.tf]: Can you please rename this variable into `cluster_encryption_config` ?

Comment (babilen5) [reply variables.tf]: Certainly. I have just pushed changes to that effect.

**Pull request merged**